x-pack/plugins/siem/public/resolver/types.ts

```diff
@@ -291,7 +291,7 @@ export type ProcessPositions = Map<ResolverEvent, Vector2>;
 /**
  * An array of vectors2 forming an polyline. Used to connect process nodes in the graph.
  */
-export type EdgeLineSegment = Vector2[];
+export type EdgeLineSegment = [Vector2, Vector2, string?];
```

Comment:
:information_source: I know this is WIP, but just to flag it early:
1) It's difficult to think of an EdgeLine segment that has "time" (which I think is the third thing) as one of its components
2) If we are doing that, I think we should adjust the comment above (`An array of vectors2 forming an polyline`)

Reply:
Yea, I initially went this way for simplicity sake, but will be refactoring it back to how it was, but have the `processEdgeLineSegments` return time as a second parameter. Generally though, I do think that its' more extensible for `EdgeLineSegments` to be stores of some type of arbitrary information that expresses the relationship between the two points. Probably not a string value, but yea, not necessary at this moment though to make that change